.clj-kondo/manifold/hooks.clj

```diff
@@ -0,0 +1,100 @@
+(ns manifold.hooks
+  (:require [clj-kondo.hooks-api :as api]))
+
+(defn- cons-vector-node
+  [node parent]
+  (api/vector-node (cons node (:children parent))))
+
+(defn def-sink-or-source [call]
+  (let [[name bindings & body] (-> call :node :children rest)
+        extended-bindings
+        (cons-vector-node (api/token-node 'lock) bindings)]
+
+    {:node
+     (api/list-node
+      (list
+       (api/token-node 'do)
+
+       (api/list-node
+        (list*
+         (api/token-node 'deftype)
+         name
+         extended-bindings
+         body))
+
+       (api/list-node
+        (list
+         (api/token-node 'defn)
+         (api/token-node (symbol (str "->" (:string-value name))))
+         bindings
+         ))))}))
+
+(defn- seq-node? [node]
+  (or (api/vector-node? node)
+      (api/list-node? node)))
+
+(defn- nth-child [node n] (nth (:children node) n))
+
+(defn both [call]
+  (let [body (-> call :node :children second :children)]
+
+    {:node
+     (api/list-node
+      (list
+       (api/token-node 'do)
+
+       (api/list-node
+        (->> body
+             (mapcat
+              #(if (and (seq-node? %) (= 'either (:value (nth-child % 0))))
+                 (:children (nth-child % 1))
+                 [%]))))
+
+       (api/list-node
+        (->> body
+             (mapcat
+              #(if (and (seq-node? %) (= 'either (:value (nth-child % 0))))
+                 (:children (nth-child % 2))
+                 [%]))))))}))
```

Comment:
Anonymous fns with the reader literal (`#()`) are best suited for one-liners.

Consider using `(fn [node] ...)`. 

Or, since they're identical, extract into a shared fn with a `child-num` param or somesuch, and pass in `(partial my-extracted-fn 1)` and `(partial my-extracted-fn 2)` to mapcat.